### PR TITLE
Fix `Index out of range` unexpected exception

### PR DESCRIPTION
**Pull request by kasperski95**
There's a desynchronization between the number of events collected in `CheatableSysCallHandler.internal_calls.events` and the event's order.

https://discord.com/channels/793094838509764618/964116575387021413/1019224706882215956

### CODE REVIEW

**Review by Arcticae — APPROVED** [review]

File: tests/integration/conftest.py [hunk 1]
```diff
@@ -112,6 +113,7 @@ async def run_cairo_test_runner(
         seed: Optional[int] = None,
         disable_hint_validation=False,
         cairo_path: Optional[List[Path]] = None,
+        test_cases: Optional[List[str]] = None,
```
Comment (mkaput) [inline tests/integration/conftest.py:116]: dead code?

File: tests/integration/cheatcodes/deploy_contract/deploy_contract_test.cairo [hunk 2]
```diff
@@ -256,3 +256,25 @@ func test_constructor_no_args_executed{syscall_ptr: felt*, range_check_ptr}() {
...
+}
+
+@event
+func fake_event() {
+}
+
+
+@external
+func test_emitting_events_from_user_contract_constructor_and_from_current_contract{syscall_ptr: felt*, range_check_ptr}() {
+    // https://github.com/software-mansion/protostar/issues/823
```
Comment (mkaput) [inline tests/integration/cheatcodes/deploy_contract/deploy_contract_test.cairo:273]: FYI that's how I'd write this, but this is fine and there's no need to change :)

```suggestion
// https://github.com/software-mansion/protostar/issues/823
@external
func test_issue_823{syscall_ptr: felt*, range_check_ptr}() {
```

**Review by mkaput — APPROVED** [review]

File: tests/integration/conftest.py (hunk 1 above)
Comment (kasperski95) [reply tests/integration/conftest.py:116]: I use it to debug a specific test case.

File: tests/integration/cheatcodes/deploy_contract/deploy_contract_test.cairo (hunk 2 above)
Comment (kasperski95) [reply tests/integration/cheatcodes/deploy_contract/deploy_contract_test.cairo:273]: I'd rather keep a descriptive name for a test case. The link could be dead in the future.

**Pull request merged**